src/test/java/io/cdap/plugin/servicenow/source/ServiceNowInputFormatTest.java

```diff
@@ -17,79 +17,180 @@
   }
 
   @Test
-  public void testFetchTableInfoReportingMode() {
+  public void testFetchTableInfoReportingMode() throws Exception {
     SourceQueryMode mode = SourceQueryMode.REPORTING;
-    ServiceNowColumn column1 = new ServiceNowColumn("sys_created_by", "string");
-    ServiceNowColumn column2 = new ServiceNowColumn("sys_updated_by", "string");
-    List<ServiceNowColumn> columns = new ArrayList<>();
-    columns.add(column1);
-    columns.add(column2);
-    ServiceNowTableDataResponse response = new ServiceNowTableDataResponse();
-    response.setColumns(columns);
-    assertEquals(3, ServiceNowInputFormat.fetchTableInfo(mode, config).size());
-    assertTrue("TABLE", true);
+    ServiceNowTableAPIClientImpl restApi = Mockito.mock(ServiceNowTableAPIClientImpl.class);
+    PowerMockito.whenNew(ServiceNowTableAPIClientImpl.class).
+            withArguments(Mockito.any(ServiceNowBaseSourceConfig.class)).thenReturn(restApi);
+    List<Map<String, Object>> result = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put("key", "value");
+    result.add(map);
+    int httpStatus = HttpStatus.SC_OK;
+    Map<String, String> headers = new HashMap<>();
+    String responseBody = "{\n" +
+            "    \"result\": []\n" +
+            "}";
+    RestAPIResponse restAPIResponse = new RestAPIResponse(httpStatus, headers, responseBody);
+    Mockito.when(restApi.executeGet(Mockito.any())).thenReturn(restAPIResponse);
+    Mockito.when(restApi.parseResponseToResultListOfMap(restAPIResponse.getResponseBody())).thenReturn(result);
+    OAuthClient oAuthClient = Mockito.mock(OAuthClient.class);
+    PowerMockito.whenNew(OAuthClient.class).
+            withArguments(Mockito.any(URLConnectionClient.class)).thenReturn(oAuthClient);
+    OAuthJSONAccessTokenResponse accessTokenResponse = Mockito.mock(OAuthJSONAccessTokenResponse.class);
+    Mockito.when(oAuthClient.accessToken(Mockito.any(), Mockito.anyString(), Mockito.any(Class.class))).
+            thenReturn(accessTokenResponse);
+    Mockito.when(accessTokenResponse.getAccessToken()).thenReturn("token");
+    RestAPIResponse response = Mockito.spy(restAPIResponse);
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    HttpClientBuilder httpClientBuilder = Mockito.mock(HttpClientBuilder.class);
+    PowerMockito.mockStatic(HttpClientBuilder.class);
+    PowerMockito.mockStatic(RestAPIResponse.class);
+    PowerMockito.when(HttpClientBuilder.create()).thenReturn(httpClientBuilder);
+    Mockito.when(httpClientBuilder.build()).thenReturn(httpClient);
+    CloseableHttpResponse httpResponse = Mockito.mock(CloseableHttpResponse.class);
+    Mockito.when(httpClient.execute(Mockito.any())).thenReturn(httpResponse);
+    PowerMockito.when(RestAPIResponse.parse(ArgumentMatchers.any(), ArgumentMatchers.anyString())).
+            thenReturn(response);
+
+    assertEquals(0, ServiceNowInputFormat.fetchTableInfo(mode, config).size());
```

Comment:
Same comment as above

Reply:
Added the new result and updated the assertion. c0e6a0f